pkg/handlers/generic/lifecycle/cpi/aws/handler.go

```diff
@@ -0,0 +1,180 @@
+			"1.27": "aws-cpi-v1.27.1",
+		},
+		"map of provider cluster implementation type to default installation ConfigMap name",
+	)
+}
+
+type AWSCPI struct {
+	client ctrlclient.Client
+	config *AWSCPIConfig
+}
+
+func New(
+	c ctrlclient.Client,
+	cfg *AWSCPIConfig,
+) *AWSCPI {
+	return &AWSCPI{
+		client: c,
+		config: cfg,
+	}
+}
+
+func (a *AWSCPI) EnsureCPIConfigMapForCluster(
+	ctx context.Context,
+	cluster *clusterv1.Cluster,
+) (*corev1.ConfigMap, error) {
+	log := ctrl.LoggerFrom(ctx).WithValues(
+		"cluster",
+		cluster.Name,
+	)
+	log.Info("Creating AWS CPI ConfigMap for Cluster")
+	version, err := semver.New(cluster.Spec.Topology.Version)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse version from cluster %w", err)
+	}
+	minorVersion := fmt.Sprintf("%d.%d", version.Major, version.Minor)
+	configMapForMinorVersion := a.config.kubernetesMinorVersionToCPIConfigMapNames[minorVersion]
+	cpiConfigMapForMinorVersion := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: a.config.defaultsNamespace,
+			Name:      configMapForMinorVersion,
+		},
+	}
+	objName := ctrlclient.ObjectKeyFromObject(
+		cpiConfigMapForMinorVersion,
+	)
+	err = a.client.Get(ctx, objName, cpiConfigMapForMinorVersion)
+	if err != nil {
+		log.Error(err, "failed to fetch CPI template for cluster")
+		return nil, fmt.Errorf(
+			"failed to retrieve default AWS CPI manifests ConfigMap %q: %w",
+			objName,
+			err,
+		)
+	}
+
+	cpiConfigMap, err := generateCPIConfigMapForCluster(ctx, cpiConfigMapForMinorVersion, cluster)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to apply AWS CPI manifests ConfigMap: %w",
+			err,
+		)
+
+	}
+	if err := client.ServerSideApply(ctx, a.client, cpiConfigMap); err != nil {
+		log.Error(err, "failed to apply CPI configmap for cluster")
+		return nil, fmt.Errorf(
+			"failed to apply AWS CPI manifests ConfigMap: %w",
+			err,
+		)
+	}
+	return cpiConfigMap, nil
+}
+
+func generateCPIConfigMapForCluster(
+	ctx context.Context,
+	cpiConfigMapForVersion *corev1.ConfigMap, cluster *clusterv1.Cluster,
+) (*corev1.ConfigMap, error) {
+	log := ctrl.LoggerFrom(ctx).WithValues(
+		"cluster",
+		cluster.Name,
+	)
+	for k, contents := range cpiConfigMapForVersion.Data {
+		objs, err := utilyaml.ToUnstructured([]byte(contents))
+		if err != nil {
+			log.Error(err, "failed to parse yaml")
+			continue
+		}
+		for i := range objs {
+			obj := objs[i]
+			if obj.GetKind() == kindDaemonset {
+				cpiDaemonSet := &appsv1.DaemonSet{}
+				err = runtime.DefaultUnstructuredConverter.FromUnstructured(
+					obj.UnstructuredContent(),
+					cpiDaemonSet,
+				)
+				if err != nil {
+					log.Error(err, "failed to convert unstructured to DaemonSet")
+					return nil, fmt.Errorf("failed to convert unstructured to DaemonSet %w", err)
+				}
+				cpiDaemonSet.Spec.Template.Spec.Containers[0].Args = append(
+					cpiDaemonSet.Spec.Template.Spec.Containers[0].Args,
+					fmt.Sprintf("--cluster-name=%s", cluster.Name),
+				)
```

Comment:
I'm not seeing `--cluster-name` in the example [specs](https://github.com/kubernetes/cloud-provider-aws/blob/a76e0ad4e9e0ff6bc8b09991ff97bf9e2cc10a77/examples/existing-cluster/base/aws-cloud-controller-manager-daemonset.yaml#L32-L38) or the helm chart.

Can you please try without it, not needing this would really help removing all this code here.

Reply:
This was taken from our konvoy2 code https://github.com/mesosphere/konvoy2/blob/main/pkg/capi/generator/templates/cpi/aws-cpi-manifests.yaml#L159

Reply:
removed and it seems to work.